autofs-5.0.7 - fix possible use after free in lookup_dir.c:lookup_init()

From: Ian Kent <raven@themaw.net>

Add a missing error return in lookup_dir.c:lookup_init().
---
 modules/lookup_dir.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/lookup_dir.c b/modules/lookup_dir.c
index 07471b7..cbeda1f 100644
--- a/modules/lookup_dir.c
+++ b/modules/lookup_dir.c
@@ -98,6 +98,7 @@ int lookup_init(const char *mapfmt, int argc, const char *const *argv, void **co
 		free(ctxt);
 		warn(LOGOPT_NONE, MODPREFIX
 		     "dir map %s, is not a directory", argv[0]);
+		return 1;
 	}
 
 	*context = ctxt;
